partiql-planner/src/main/kotlin/org/partiql/planner/metadata/Scope.kt

```diff
@@ -0,0 +1,61 @@
+package org.partiql.planner.metadata
+
+/**
+ * A [Scope] is a namespace for the following objects; top-level scopes are called "Catalogs".
+ *
+ * . Tables
+ *      . Base
+ *      . View
+ *      . Index
+ * . Routines
+ *      . Functions
+ *      . Procedures
+ * . Scopes
+ * . Types
+ *
+ * See, https://github.com/apache/calcite/blob/main/core/src/main/java/org/apache/calcite/schema/Schema.java
+ */
+public interface Scope {
+
+    /**
+     * The [Scope] name.
+     */
+    public fun getName(): String
+
+    /**
+     * Get a table by name.
+     *
+     * @param name  The case-sensitive [Table] name.
+     * @return      The [Table] or null if not found.
+     */
+    public fun getTable(name: String): Table? = null
```

Comment:
For this method (as well as the ones below), I'm wondering how we can determine the case-sensitive names for third-party consumers (that may not consistently upper-or-lower-case their table names upon storage).

Reply:
We don't determine the case-sensitive names; I think you have consumer-implementer backwards. We case-normalize to lowercase and it's up to the implementer how to handle.

In other words, all usage of the Scope comes from the planner; so the planner is only "consumer" of Scope — implementers are responsible for handling their own mapping from _PartiQL's case-normalized_